### PR TITLE
TSK-1064: removed DeclarationOrderCheck from checkstyle

### PR DESCRIPTION
**Pull request by mustaphazorgati**

### CODE REVIEW

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import java.time.Duration;
```
Comment (mustaphazorgati) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: This entire file was just refactored and "technically" has nothing to do with this PR. I just stumpled over it and had the time to fix it :)

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 2]
```diff
@@ -733,9 +732,8 @@ Duration calculateDuration(
       return result;
     }
 
-    Set<String> taskIdSet =
-        taskSummaries.stream().map(TaskSummaryImpl::getTaskId).collect(Collectors.toSet());
-    String[] taskIdArray = taskIdSet.toArray(new String[0]);
+    String[] taskIdArray =
+        taskSummaries.stream().map(TaskSummaryImpl::getTaskId).distinct().toArray(String[]::new);
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: 👍

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 3]
```diff
@@ -982,26 +980,17 @@ private boolean desiredCallbackStateCanBeSetForFoundSummary(
 
     switch (desiredCallbackState) {
       case CALLBACK_PROCESSING_COMPLETED:
-        if (!(currentTaskState.equals(TaskState.COMPLETED))) {
-          return false;
-        }
-        return true;
+        return currentTaskState.equals(TaskState.COMPLETED);
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: 👍

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 4]
```diff
@@ -1239,7 +1228,7 @@ private void addClassificationSummariesToTaskSummaries(
       TaskImpl task, List<AttachmentImpl> attachmentImpls) {
     LOGGER.debug("entry to transferBulk()");
     Set<String> classificationIdSet =
-        new HashSet<>(Arrays.asList(task.getClassificationSummary().getId()));
+        new HashSet<>(Collections.singletonList(task.getClassificationSummary().getId()));
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: 👍

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 5]
```diff
@@ -1269,11 +1258,11 @@ private void addWorkbasketSummariesToTaskSummaries(List<TaskSummaryImpl> taskSum
...
     }
     // calculate parameters for workbasket query: workbasket keys
-    Set<String> workbasketIdSet =
+    String[] workbasketIdArray =
         taskSummaries.stream()
             .map(t -> t.getWorkbasketSummary().getId())
-            .collect(Collectors.toSet());
-    String[] workbasketIdArray = workbasketIdSet.toArray(new String[0]);
+            .distinct()
+            .toArray(String[]::new);
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: 👍

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 6]
```diff
@@ -1429,7 +1418,7 @@ private void validateCustomFields(Map<String, String> customFieldsToUpdate)
       throws InvalidArgumentException {
     if (LOGGER.isDebugEnabled()) {
       LOGGER.debug(
-          "entry to validateCustomFields(customFieldsToUpdate = {}, taskIds = {})",
+          "entry to validateCustomFields(customFieldsToUpdate = {})",
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: 👍

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java [hunk 7]
```diff
@@ -1980,7 +1968,6 @@ private void createTasksCompletedEvents(List<TaskSummary> taskSummaries) {
     private int prio;
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: without the setter you can make prio final. maybe you can do that for duration too? makes the whole object immutable.

File: qa/checkstyle/checkstyle.xml [hunk 8]
```diff
@@ -177,7 +177,6 @@
         value="Method type name ''{0}'' must match pattern ''{1}''."/>
       <property name="format" value="(^[A-Z][0-9]?)$|([A-Z][a-zA-Z0-9]*[T]$)"/>
     </module>
-    <module name="DeclarationOrderCheck"/>
```
Comment (benjamineckstein) [inline qa/checkstyle/checkstyle.xml]: why do we remove this?

Comment (mustaphazorgati) [reply qa/checkstyle/checkstyle.xml]: see ticket :)
TL;DR - java formatter has different ordering than that [checkstyle config](https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/coding/DeclarationOrderCheck.html).

From checkstyle:
> According to Code Conventions for the Java Programming Language, the parts of a class or interface declaration should appear in the following order:

Class (static) variables. First the public class variables, then protected, then package level (no access modifier), and then private.
Instance variables. First the public class variables, then protected, then package level (no access modifier), and then private.
Constructors
Methods

Comment (mustaphazorgati) [reply qa/checkstyle/checkstyle.xml]: our formatter behaves differently.

File: lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java (hunk 7 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java]: good point :)

**Comment by mustaphazorgati** [comment]
@benjamineckstein am I stupid? Why the heck is the build broken? Any idea?

**Comment by mustaphazorgati** [comment]
Because of #893. Will reopen after new structure :)
